### ISSUE TITLE
Entropy loss error ?

### ISSUE THREAD

**Issue by gouldglenn** [dreamquark-ai/tabnet#311]
Hi all, first thanks for making this code available, learning a lot of stuff there
Looks like there a missing negative sign in the entropy loss (doesn't match paper's formula):

I guess it's more:
```
M_loss += torch.mean(
                torch.sum(torch.mul(-M, torch.log(M + self.epsilon)), dim=1)
            )
```

https://github.com/dreamquark-ai/tabnet/blob/b6e1ebaf694f37ad40a6ba525aa016fd3cec15da/pytorch_tabnet/tab_network.py#L162

Cheers

**Comment by Optimox** [comment]
Hello, I’m on holidays so can’t spend too much time looking at this, but I’m pretty sure that we would have noticed if the loss was of the wrong sign ^^
Isn’t the minus sign inside the torch.mul before the M?

**Comment by gouldglenn** [comment]
Yeah actually the negative sign happens somewhere else in the code : 

https://github.com/dreamquark-ai/tabnet/blob/b6e1ebaf694f37ad40a6ba525aa016fd3cec15da/pytorch_tabnet/abstract_model.py#L475

Sorry to have interrupted your holiday, enjoy !
